consensus/actor_state.go

```diff
@@ -29,11 +29,13 @@ type ActorStateStore struct {
 	cst *hamt.CborIpldStore
 	// For vm storage.
 	bs bstore.Blockstore
+	// executable actors
+	processor Processor
```

Comment:
I'm not convinced this is a desirable dependency. I gather it's introduced by removing the actor code map from the state tree? I do see the code was already invoking `CallQueryMethod` from the processor. Maybe we need an abstraction that wraps up the state tree *and* the code table (for want of a better term), i.e. *all* the "memory" of the VM.

No immediate action required.

Reply:
I'm not either, but here's the motivation: builtin actors and the protocol version table are required for protocol version variant message execution. These objects are singleton for the execution lifetime of the app and `ActorState` doesn't use them directly. It simply passes them on to processor whenever it calls Query. Other code running state transitions already keeps a reference to the processor to ApplyMessages. This change standardizes patterns of usage while reducing `ActorState`s exposure to execution details.

I'll work towards the cleaner abstraction you mention in the next PR.

Reply:
I'm not either, but here's the motivation: Without giving ActorState a reference to processor, it would now be forced to keep a reference to bulitin.Actors and pass it on to the Query method without actually using it. This will soon be true for the ProtocolVersionTable (or something containing it) as well. Other code that runs state transitions are already injected with a reference to Protocol to call ApplyMessage and its variants. This change standardizes patterns of usage while reducing `ActorStates` exposure to execution details.

Your idea for a new abstraction is a good one, and i'll be working towards that in the next PR.